### PR TITLE
My solution to the "Make it Roll" assignment

### PR DESCRIPTION
**Pull request by fbottarel**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -68,38 +72,83 @@ class CtrlModule: public RFModule
...
         // FILL IN THE CODE
-        return Vector(4);
+        // right hand palm is parallel to the xz plane
+        // specify rotation matrix wrt root frame and convert it
+        Matrix pose_mat(3,3);
+        pose_mat(0,0)=-1.0;  pose_mat(0,1)= 0.0;  pose_mat(0,2)= 0.0;    // x
+        pose_mat(1,0)= 0.0;  pose_mat(1,1)= 0.0;  pose_mat(1,2)=-1.0;    // y
+        pose_mat(2,0)= 0.0;  pose_mat(2,1)=-1.0;  pose_mat(2,2)= 0.0;    // z
+
+        return yarp::math::dcm2axis(pose_mat);
```
Comment (giuliavezzani) [inline src/main.cpp]: It's not necessary to use the namespace `yarp::math` because of line 16.

File: src/main.cpp [hunk 2]
```diff
@@ -127,6 +176,19 @@ class CtrlModule: public RFModule
     void home()
     {
         // FILL IN THE CODE
+        // restore pose and gaze
+        // fix torso DOF
+        iarm    -> setLimits(0, -0.1, 0.1);
```
Comment (giuliavezzani) [inline src/main.cpp]: As @pattacini told you, you could use also:
```
iarm    -> setLimits(0, 0.0, 0.0);
iarm    -> setLimits(1, 0.0, 0.0);
iarm    -> setLimits(2, 0.0, 0.0);
```

The solver is able to handle this constraints.

File: src/main.cpp [hunk 3]
```diff
@@ -162,12 +224,75 @@ class CtrlModule: public RFModule
...
+
+        if (!ok)
+        {
+            yError() << "Unable to open Gaze Controller";
+            return false;
+        }
+
+        // drivers are up, open views now
+        drvArm.view(iarm);
+        drvGaze.view(igaze);false;   // you need to properly handle this flag
```
Comment (giuliavezzani) [inline src/main.cpp]: I think  `false;` at line 255 is a typo.

File: src/main.cpp (hunk 1 above)
Comment (fbottarel) [reply src/main.cpp]: Indeed!

File: src/main.cpp (hunk 2 above)
Comment (fbottarel) [reply src/main.cpp]: That was the idea, but I wasn sure! 
Fixed.

File: src/main.cpp (hunk 3 above)
Comment (fbottarel) [reply src/main.cpp]: Typo indeed.

**Pull request merged**